channels/layers.py

```diff
@@ -329,22 +330,29 @@ async def group_discard(self, group, channel):
         assert self.valid_channel_name(channel), "Invalid channel name"
         assert self.valid_group_name(group), "Invalid group name"
         # Remove from group set
-        if group in self.groups:
-            if channel in self.groups[group]:
-                del self.groups[group][channel]
-            if not self.groups[group]:
-                del self.groups[group]
+        group_channels = self.groups.get(group, None)
+        if group_channels:
+            # remove channel if in group
+            group_channels.pop(channel, None)
+            # is group now empty? If yes remove it
+            if not group_channels:
+                self.groups.pop(group, None)
 
     async def group_send(self, group, message):
         # Check types
         assert isinstance(message, dict), "Message is not a dict"
         assert self.valid_group_name(group), "Invalid group name"
         # Run clean
         self._clean_expired()
+
         # Send to each channel
-        for channel in self.groups.get(group, set()):
+        ops = []
+        if group in self.groups:
+            for channel in self.groups[group].keys():
+                ops.append(asyncio.ensure_future(self.send(channel, message)))
```

Comment:
Isn't it better to use `asyncio.create_task` instead of `asyncio.ensure_future`? Shouldn't we use the higher-level mechanism (as recommended in the docs)? Or is there a good reason to do it this way that I'm not seeing?

Reply:
lowlevel is faster, but not sure about this.

Reply:
How much faster? Unless it's WAY faster I'm not sure it's worth going against the docs here.

Reply:
it was a year ago, so most probably not worth the effort. Changed to create_task